src/screens/Notifications/ConfirmEmailScreen.tsx

```diff
@@ -149,12 +154,15 @@ class ConfirmEmailScreen extends Component<Props, State> {
   };
 
   onConfirm = () => {
-    const { sessionToken } = this.props;
+    const {
+      sessionToken,
+      route: { params },
+    } = this.props;
     const { code } = this.state;
 
     this.props.authenticate(sessionToken, code, {
       onFailure: this.onError,
-      onSuccess: this.infoContainerContent.onCodeConfirm,
+      onSuccess: this.infoContainerContent.onCodeConfirm || params.onSuccess,
```

Comment:
please don't mix technics and patterns, either refactor all flows so that they will pass onSuccess method as params, or add a new flowType. I vote for the first options, props based components are the most flexible technic for me

Reply:
I added flowType, I will refactor it in the next PR not to block testers from giving feedback on the update flow

Reply:
ok, please delete params.onSuccess in this one